infinispan/runtime/src/main/java/io/quarkiverse/langchain4j/infinispan/InfinispanEmbeddingStore.java

```diff
@@ -0,0 +1,176 @@
+        return id;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream()
+                .map(ignored -> randomUUID())
+                .collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment embedded) {
+        addAllInternal(singletonList(id), singletonList(embedding), embedded == null ? null : singletonList(embedded));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> embedded) {
+        if (ids.isEmpty() || ids.size() != embeddings.size() || (embedded != null && embedded.size() != embeddings.size())) {
+            throw new IllegalArgumentException("ids, embeddings and embedded must be non-empty and of the same size");
+        }
+        ensureTrue(ids.size() == embeddings.size(), "ids size is not equal to embeddings size");
+        ensureTrue(embedded == null || embeddings.size() == embedded.size(), "embeddings size is not equal to embedded size");
+
+        int size = ids.size();
+        Map<String, LangchainInfinispanItem> elements = new HashMap<>(size);
+        for (int i = 0; i < size; i++) {
+            String id = ids.get(i);
+            Embedding embedding = embeddings.get(i);
+            TextSegment textSegment = embedded == null ? null : embedded.get(i);
+            if (textSegment != null) {
+                Map<String, String> map = textSegment.metadata().asMap();
+                final List<String> metadataKeys = new ArrayList<>(map.size());
+                final List<String> metadataValues = new ArrayList<>(map.size());
+                map.entrySet().forEach(e -> {
+                    metadataKeys.add(e.getKey());
+                    metadataValues.add(e.getValue());
+                });
+                elements.put(id,
+                        new LangchainInfinispanItem(id, embedding.vector(), textSegment.text(), metadataKeys, metadataValues));
+            } else {
+                elements.put(id, new LangchainInfinispanItem(id, embedding.vector(), null, null, null));
+            }
+        }
+        // blocking call
+        remoteCache.putAll(elements);
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults,
+            double minScore) {
+        Query<Object[]> query = remoteCache.query("select i, score(i) from " + "LanchainItem" + schema.getDimension().toString()
+                + " i where i.floatVector <-> " + Arrays.toString(referenceEmbedding.vector()) + "~3");
```

Comment:
What does the `~3` mean? Maximum vector distance or something? Should it be configurable rather than hardcoded?